streams/src/main/java/org/apache/kafka/streams/processor/assignment/TaskAssignmentUtils.java

```diff
@@ -613,9 +704,17 @@ private static Map<ProcessId, KafkaStreamsAssignment> loadBasedStandbyTaskAssign
             .collect(Collectors.toSet());
         final Map<TaskId, Integer> tasksToRemainingStandbys = statefulTaskIds.stream()
             .collect(Collectors.toMap(Function.identity(), t -> numStandbyReplicas));
-        final Map<UUID, KafkaStreamsAssignment> clients = kafkaStreamsAssignments.entrySet().stream().collect(Collectors.toMap(
+        final Map<UUID, KafkaStreamsAssignment> clients = streamStates.entrySet().stream().collect(Collectors.toMap(
             entry -> entry.getKey().id(),
-            Map.Entry::getValue
+            entry -> {
+                final KafkaStreamsState state = entry.getValue();
+                if (kafkaStreamsAssignments.containsKey(state.processId())) {
+                    return kafkaStreamsAssignments.get(state.processId());
+                }
+                final KafkaStreamsAssignment newAssignment = KafkaStreamsAssignment.of(state.processId(), new HashSet<>());
+                kafkaStreamsAssignments.put(state.processId(), newAssignment);
+                return newAssignment;
+            }
```

Comment:
IIUC this is essentially the same "bug" as above, ie accounting for an invalid assignment being passed in where not all ProcessIds in the KafkaStreamsState map have a corresponding KafkaStreamsAssignment, right? 

It feels a little "sneaky" to just fill in missing assignments while constructing this map. Also, it seems like we'd want to do the same thing -- initialize any missing ProcessIds with empty an KafkaStreamsAssignment -- for each of the methods here, ie #defaultStandbyTaskAssignment and the two rack-aware optimization methods for active & standby tasks. So maybe just split out this logic into a separate helper function that each method can call at the initialization period before the actual algorithm begins. That way we don't need to worry about this ever again

Also: once we do the UUID->ProcessID thing, we'll want to get rid of this particular variable altogether since it's only purpose in the beginning was to convert from a Map<ProcessId, KafkaStreamsAssignment> to a Map<UUID, KafkaStreamsAssignment>. We don't want to add more logic here specifically